### PR TITLE
ADEN-4640 Handle resolve state ads

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/slot/resolveState.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/* global define */
+define('ext.wikia.adEngine.slot.resolveState', [
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/slot/resolveState.js:2]: What do you think about moving it to some UAP related directory?

File: extensions/wikia/AdEngine/js/slot/resolveState.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		return params.resolveState.aspectRatio > 0 && params.resolveState.imageSrc !== '';
+	}
+
+	function hasResolvedState(params) {
+		return isForcedByURLParam() || (paramsAreCorrect(params) && !isBlockedByURLParam());
+	}
+
+	function updateAd(params) {
+		log('Resolve state is turned on', logGroup, log.levels.debug);
+		replaceImage(params);
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/slot/resolveState.js]: Does it mean that ad starts with big image? What do you think about setting urls in both cases and get rid of additional request for image when resolve state is called

File: extensions/wikia/AdEngine/js/slot/resolveState.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	function replaceImage(params) {
+		params.backgroundImage.src = params.resolveState.imageSrc;
+	}
+
+	function getQueryParam() {
+		return qs.getVal('resolvedState', null);
+	}
+
+	function isForcedByURLParam() {
+		return [true, 'true'].indexOf(getQueryParam()) > -1;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/slot/resolveState.js]: Could you add `'1'` value?

File: extensions/wikia/AdEngine/js/slot/resolveState.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	function isBlockedByURLParam() {
+		return [false, 'blocked', 'false'].indexOf(getQueryParam()) > -1;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/slot/resolveState.js]: Could you add `'0'` value?

File: extensions/wikia/AdEngine/js/slot/resolveState.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+	var logGroup = 'ext.wikia.adEngine.slot.resolveState',
+		qs = new QueryString();
+
+	function replaceImage(params) {
+		params.backgroundImage.src = params.resolveState.imageSrc;
+	}
+
+	function getQueryParam() {
+		return qs.getVal('resolvedState', null);
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/slot/resolveState.js]: Can we use underscore?

File: extensions/wikia/AdEngine/js/slot/resolveState.js (hunk 1 above)
Comment (v3rth) [reply extensions/wikia/AdEngine/js/slot/resolveState.js:2]: discussed offline, there is no good place for that function :(

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**